datadog/dogstatsd/base.py

```diff
@@ -488,23 +488,24 @@ def _is_telemetry_flush_time(self):
         return False
 
     def _send_to_server(self, packet):
-        flush_telemetry = self._is_telemetry_flush_time()
-        if flush_telemetry:
-            telemetry_packet = self._flush_telemetry()
-            if not self._dedicated_telemetry_destination():
-                packet += "\n"+telemetry_packet
-
         try:
+            flush_telemetry = self._is_telemetry_flush_time()
+            if flush_telemetry:
+                telemetry_packet = self._flush_telemetry()
+                if not self._dedicated_telemetry_destination():
+                    packet += "\n"+telemetry_packet
+                else:
+                    (self.telemetry_socket or self.get_socket(telemetry=True)).send(
+                        telemetry_packet.encode(self.encoding))
+
+                self._reset_telemetry()
```

Comment:
In case we don't have a _dedicated_telemetry_destination we need to wait for the `send` line 504 before reseting the telemetry or we're going to lose data (if we have network error twice in a row).